### PR TITLE
Test ProvidedRequiredArguments

### PR DESCRIPTION
**Pull request by PascalSenn**
Ports validation tests from GraphQLJs for rule ProvidedRequiredArguments

### CODE REVIEW

File: src/HotChocolate/Core/test/Validation.Tests/__snapshots__/RequiredArgumentRuleTests.IgnoresUnknownArguments.snap [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿[
+  {
+    "Message": "The argument `unknownArgument` does not exist.",
```
Comment (michaelstaib) [inline src/HotChocolate/Core/test/Validation.Tests/__snapshots__/RequiredArgumentRuleTests.IgnoresUnknownArguments.snap:3]: to what test does this snapshot belong to?

Comment (PascalSenn) [reply src/HotChocolate/Core/test/Validation.Tests/__snapshots__/RequiredArgumentRuleTests.IgnoresUnknownArguments.snap:3]: Good catch.. this one is not needed

Comment (michaelstaib) [reply src/HotChocolate/Core/test/Validation.Tests/__snapshots__/RequiredArgumentRuleTests.IgnoresUnknownArguments.snap:3]: Can we remove it or is it used?

Comment (PascalSenn) [reply src/HotChocolate/Core/test/Validation.Tests/__snapshots__/RequiredArgumentRuleTests.IgnoresUnknownArguments.snap]: Done

**Pull request merged**